lua/lualine/config.lua

```diff
@@ -6,6 +6,265 @@ local modules = require('lualine_require').lazy_require {
   utils_notices = 'lualine.utils.notices',
 }
 
+---@alias LualineThemes
+---| "auto"
+---| "16color"
+---| "ayu_dark"
+---| "ayu_light"
+---| "ayu_mirage"
+---| "ayu"
+---| "codedark"
+---| "dracula"
+---| "everforest"
+---| "gruvbox_dark"
+---| "gruvbox_light"
+---| "gruvbox"
+---| "gruvbox-material"
+---| "horizon"
+---| "iceberg_dark"
+---| "iceberg_light"
+---| "iceberg"
+---| "jellybeans"
+---| "material"
+---| "modus-vivendi"
+---| "molokai"
+---| "moonfly"
+---| "nightfly"
+---| "nord"
+---| "OceanicNext"
+---| "onedark"
+---| "onelight"
+---| "palenight"
+---| "papercolor_dark"
+---| "papercolor_light"
+---| "PaperColor"
+---| "powerline"
+---| "powerline_dark"
+---| "pywal"
+---| "seoul256"
+---| "solarized_dark"
+---| "solarized_light"
+---| "Tomorrow"
+---| "wombat"
+
+---@class LualineDisabledFiletypes
+---only ignores the ft for statusline
+---@field statusline table
+---only ignores the ft for winbar.
+---@field winbar table
+
+---@class LualineRefresh
+---@field statusline integer
+---@field tabline integer
+---@field winbar integer
+
+---@class LualineOpts
+-- Show icons.
+---@field icons_enabled boolean
+-- Lualine theme name.
+-- Auto theme will load the theme automaically for your colorscheme
+-- or will generate one otherwise.
+---@field theme LualineThemes
+-- Separators between components.
+---@field component_separators table|string
+-- Separators between sections.
+---@field section_separators table|string
+-- Disable lualine for these filetypes.
+---@field disabled_filetypes LualineDisabledFiletypes
+-- When set to true, left sections i.e. 'a','b' and 'c'
+-- can't take over the entire statusline even
+-- if neither of 'x', 'y' or 'z' are present.
+---@field always_divide_middle boolean
+---@field globalstatus boolean
+---@field refresh LualineRefresh
+
+---@alias LualineComponents
+---| "branch"
+---| "buffers"
+---| "diagnostics"
+---| "diff"
+---| "encoding"
+---| "fileformat"
+---| "filename"
+---| "filesize"
+---| "filetype"
+---| "hostname"
+---| "location"
+---| "mode"
+---| "progress`"
+---| "tabs"
+
+---@class LualineGeneralComponent
+-- Must be written as array value
+---@field [1] LualineComponents
+-- Enables the display of icons alongside the component.
+---@field icons_enabled boolean
+-- Defines the icon to be displayed in front of the component.
+---@field icon string
+-- When a string is provided it's treated as component_separator.
+-- When a table is provided it's treated as section_separator.
+-- Passing an empty string disables the separator.
+---@field separator string|table
+-- Condition function, the component is loaded when the function returns `true`.
+---@field cond fun():boolean
+-- Defines a custom color for the component.
+-- example: { fg = '#ffaa88', bg = 'grey', gui='italic,bold' } | { fg = 204 } | "WarningMsg"
+---@field color string|table
```

Comment:
create a type for color instead of table that has members fg, bg, gui. also this color option can be a function too returning a color type(table) | string.

yeah lualine's options are overloaded as hell 😆